source/models/R290-generic.yaml

```diff
@@ -591,6 +603,96 @@ add:
         - lambda: |-
             if (x > 50) return {};
             return x;
+    # Register: 191
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "TL Outdoor Refrigerant Pipe Temperature"
+      id: "${devicename}_tl_outdoor_refrigerant_pipe_temperature"
+      icon: mdi:thermometer
+      register_type: holding
+      address: 191
+      value_type: S_WORD
+      unit_of_measurement: "°C"
+      device_class: temperature
+      accuracy_decimals: 0
+    # Register: 192
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "Pump Internal PWM"
+      id: "${devicename}_pump_i_pwm"
+      icon: mdi:pump
+      register_type: holding
+      address: 192
+      value_type: U_WORD
+      unit_of_measurement: "%"
+      accuracy_decimals: 1
+      filters:
+        - lambda: return x * 0.1;
+    # Register: 193
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "T9i Second PHE Inlet Temperature"
+      id: "${devicename}_t9i_second_phe_inlet_temperature"
+      icon: mdi:thermometer
+      register_type: holding
+      address: 193
+      value_type: S_WORD
+      unit_of_measurement: "°C"
+      device_class: temperature
+      accuracy_decimals: 1
+      filters:
+        - lambda: |-
+            if (x == 0x7fff) return {};
+            return x * 0.1;
+    # Register: 194
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "T9o Second PHE Outlet Temperature"
+      id: "${devicename}_t9o_second_phe_outlet_temperature"
+      icon: mdi:thermometer
+      register_type: holding
+      address: 194
+      value_type: S_WORD
+      unit_of_measurement: "°C"
+      device_class: temperature
+      accuracy_decimals: 1
+      filters:
+        - lambda: |-
+            if (x == 0x7fff) return {};
+            return x * 0.1;
+    # Register: 195
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "EXV2 Expansion Valve Openness"
+      id: "${devicename}_exv2_openness"
+      icon: mdi:valve
+      register_type: holding
+      address: 195
+      value_type: U_WORD
+      unit_of_measurement: "P"
+      accuracy_decimals: 0
+    # Register: 196
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "EXV3 Expansion Valve Openness"
+      id: "${devicename}_exv3_openness"
+      icon: mdi:valve
+      register_type: holding
+      address: 196
+      value_type: U_WORD
+      unit_of_measurement: "P"
+      accuracy_decimals: 0
+    # Register: 197
+    - platform: modbus_controller
+      modbus_controller_id: "${devicename}"
+      name: "Fan Speed Actual"
```

Comment:
There is already a fan speed, which is the speed of the fan of the outdoor unit. In the modbus mapping I have, this one is stated as “fan2 speed”. Although I do not like the “fan2” part, but if we do not have a better method to distinguish between the 2 fans, let’s use “fan2” for now, unless you have a better name that reflects this specific fan type.